### ISSUE TITLE
read_ndjson cannot extend/append Null with UInt32

### ISSUE THREAD

**Issue by s-b90** [pola-rs/polars#12882]
### Checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Reproducible example

```python
import io
import polars as pl

text1 = '{"key": [1, 2]}'
text2 = '{"key1": [1, 2]}'


file = io.StringIO(
    '\n'.join([text1 for _ in range(100)]) +
    '\n' +
    '\n'.join([text2 for _ in range(100)])
    )
pl.read_ndjson(file, schema={"key": pl.List(pl.UInt32), "key1": pl.List(pl.UInt32)})
RuntimeError: BindingsError: "data types don't match: cannot extend/append Null with UInt32"
```

### Log output

```shell
RuntimeError: BindingsError: "data types don't match: cannot extend/append Null with UInt32".
```

### Issue description

In example nd_json file has 200 rows. For column `key` last 100 values are Null, for column `key1` first 100 values are Null.
If reduce `range(100)` to `range(10)` or other small number - everything works as expected.

PS: Another strange behavior. If remove `schema` in `pl.read_ndjson(file)` then error disappear, but df has only 1 column `key`.

### Expected behavior

read_ndjson() returns df with both columns without error (with defining schema and without it).

### Installed versions

<details>

```
--------Version info---------
Polars:               0.19.19
Index type:           UInt32
Platform:             Linux-6.2.0-37-generic-x86_64-with-glibc2.35
Python:               3.11.6 (main, Oct 23 2023, 22:48:54) [GCC 11.4.0]

----Optional dependencies----
adbc_driver_manager:  <not installed>
cloudpickle:          <not installed>
connectorx:           <not installed>
deltalake:            <not installed>
fsspec:               <not installed>
gevent:               <not installed>
matplotlib:           3.8.2
numpy:                1.26.2
openpyxl:             <not installed>
pandas:               2.1.3
pyarrow:              13.0.0
pydantic:             <not installed>
pyiceberg:            <not installed>
pyxlsb:               <not installed>
sqlalchemy:           <not installed>
xlsx2csv:             <not installed>
xlsxwriter:           <not installed>
None
```

</details>

**Comment by s-b90** [comment]
Maybe related to https://github.com/pola-rs/polars/issues/11824 and https://github.com/pola-rs/polars/issues/11496

**Comment by cmdlineluser** [comment]
The largest `N` for both is `range(64)` i.e. 128 lines

https://github.com/pola-rs/polars/blob/c1f04d1b8885711eb29debed37845992a8e2705c/crates/polars-io/src/ndjson/core.rs#L216-L218

So it looks like the difference has something to do with the multithreaded version.

The 1-column result happens regardless of the number of threads, so that seems to be a different issue.

It seems to go here next but I haven't been able to figure out the source problem as of yet:

https://github.com/pola-rs/polars/blob/c1f04d1b8885711eb29debed37845992a8e2705c/crates/polars-io/src/ndjson/core.rs#L265

**Comment by maparham** [comment]
This bug seems resolved as of py-1.9.0.

**Comment by s-b90** [comment]
Yes, can confirm this. Thanks.
